src/encoded/submission_status.py

```diff
@@ -0,0 +1,148 @@
+            "error": f'Error when trying to get submission status: {e}',
+        }
+
+    return {
+        "file_sets": file_sets,
+        "total_filesets": num_total_filesets,
+    }
+
+
+def add_search_filters(search_params, filter):
+    if not filter:
+        return
+    if "fileset_status" in filter and filter["fileset_status"] != "all":
+        search_params[STATUS] = filter["fileset_status"]
+    if "submission_center" in filter:
+        if filter["submission_center"] == "all_gcc":
+            search_params["submission_centers.display_title"] = [
+                WASHU_GCC,
+                BCM_GCC,
+                NYGC_GCC,
+                BROAD_GCC,
+                UWSC_GCC,
+            ]
+        elif filter["submission_center"] != "all":
+            search_params["submission_centers.display_title"] = filter[
+                "submission_center"
+            ]
+    if "fileset_created_from" in filter and filter["fileset_created_from"]:
+        search_params["date_created.from"] = filter["fileset_created_from"]
+    if "fileset_created_to" in filter and filter["fileset_created_to"]:
+        search_params["date_created.to"] = filter["fileset_created_to"]
+
+
+def process_files_metadata(files_metadata):
+    is_upload_complete = True
+    num_files_copied_to_o2 = 0
+    file_formats = []
+    submitted_files = list(
+        filter(lambda f: SUBMITTED_FILE in f["@type"], files_metadata)
+    )
+    for file in submitted_files:
+        if file[STATUS] == UPLOADING:
+            is_upload_complete = False
+        if O2_PATH in file:
+            num_files_copied_to_o2 += 1
+        file_formats.append(file.get(FILE_FORMAT, {}).get(DISPLAY_TITLE))
+
+    # Make it unqique
+    file_formats = list(set(file_formats))
+
+    date_uploaded = None
+    if is_upload_complete and len(submitted_files) > 0:
+        for file in submitted_files:
+            file_status_tracking = file.get("file_status_tracking")
+            if file_status_tracking and UPLOADED in file_status_tracking:
+                date_uploaded_current = file_status_tracking[UPLOADED]
+                if not date_uploaded:
+                    date_uploaded = date_uploaded_current
+                    continue
+                date_uploaded = (
+                    date_uploaded_current
+                    if date_uploaded_current > date_uploaded
+                    else date_uploaded
+                )
+
+    return {
+        "is_upload_complete": is_upload_complete,
+        "num_submitted_files": len(submitted_files),
+        "date_uploaded": date_uploaded,
+        "file_formats": ", ".join(file_formats),
+        "num_files_copied_to_o2": num_files_copied_to_o2,
+    }
+
+
+def search_total(context, request, search_params):
+    """Reads search params and executes a search total"""
+    ignored(context)
+
+    # Ensure this is always the case to prevent unintended slowness
+    if "limit" not in search_params:
+        search_params["limit"] = 0
```

Comment:
This is probably not robust enough - one could still pass a bad "limit" parameter. You should ignore what is passed in for limit when getting a search total and always set it to 1.

Reply:
Good point